src/Common/src/CoreLib/System/MemoryExtensions.cs

```diff
@@ -292,31 +276,14 @@ public static bool SequenceEqual<T>(this Span<T> first, ReadOnlySpan<T> second)
         public static int SequenceCompareTo<T>(this Span<T> first, ReadOnlySpan<T> second)
             where T : IComparable<T>
         {
-            if (typeof(T) == typeof(byte))
```

Comment:
I don't think you can validly apply this optimization to `SequentialCompare` - comparing unsigned bytes one at a time isn't the same as comparing elements using the proper Compare algorithm.

Reply:
Good point. I will revert this. We need to add more tests for T != byte for SequenceCompareTo.

Reply:
https://github.com/dotnet/corefx/issues/28118